### ISSUE TITLE
Subdomain hierarchy should be displayed, even if not existing (a.b.example.com should be under b.example.com)

### ISSUE THREAD

**Issue by mathieumd** [happyDomain/happydomain#26]
This was not very clear, at first, why some RR are indented and others not:
![subdomains](https://github.com/happyDomain/happydomain/assets/2104679/3cab3230-d7ed-4020-91e1-674dee519bfa)

I believe adding a *false* `b.example.com` (greyed out or marked as nonexistent in any way, of course), would help to highlight the fact that it's the parent subdomain.

**Comment by nemunaire** [comment]
Of course, this is weird as is!

Here is what we should expect instead:
![image](https://github.com/happyDomain/happydomain/assets/682394/ef22dd59-39f2-4faa-8e36-b1b579d52929)

The link is valid and takes to the right location in the page: on the first subdomain under b.

But I wonder if people wouldn't expect to have the subdomain listed anyway in the services list, to add a service later. Like this:

![image](https://github.com/happyDomain/happydomain/assets/682394/0a744829-3555-4de0-ac1a-323b7fdba7e4)

Finally, on reverse zone, especially reverse IPv6, I think this is unwanted (too much unexpected 0.0.0.0...)

**Comment by mathieumd** [comment]
> The link is valid and takes to the right location in the page: on the first subdomain under b.

Yes, it looks good.

> But I wonder if people wouldn't expect to have the subdomain listed anyway in the services list, to add a service later.

Good idea too.

> Finally, on reverse zone, especially reverse IPv6, I think this is unwanted (too much unexpected 0.0.0.0... or duplicate info)

Sure! Maybe by folding hierarchy of empty records?

```
example.com.
  0.example.com.
  b.example.com.
    a.b.example.com.
      0.a.b.example.com.
        [v]
          5.4.3.2.1.0.a.b.example.com.
  c.example.com.
```

**Comment by nemunaire** [comment]
> > Finally, on reverse zone, especially reverse IPv6, I think this is unwanted (too much unexpected 0.0.0.0... or duplicate info)
> 
> Sure! Maybe by folding hierarchy of empty records?
> 
> ```
> example.com.
>   0.example.com.
>   b.example.com.
>     a.b.example.com.
>       0.a.b.example.com.
>         [v]
>           5.4.3.2.1.0.a.b.example.com.
>   c.example.com.
> ```

Folding is an excellent solution.

Perhaps, to avoid a dynamic content with too many intermediate subdomains, we can just keep the interesting branches:

```
example.com.
  0.example.com.
  b.example.com.
      0.a.b.example.com.
          4.4.3.2.1.0.a.b.example.com.
          5.4.3.2.1.0.a.b.example.com.
      1.a.b.example.com.
          6.5.4.3.2.1.a.b.example.com.
  c.example.com.
```

**Comment by mathieumd** [comment]
> Perhaps, to avoid a dynamic content with too many intermediate subdomains, we can just keep the interesting branches:

Yes, but displaying both the first and last nonexistent RR before the existent RR seems more readable, no?
And using a hierarchy symbol too (here copied from `tree`).

```
example.com.
├─ 0.example.com.
├─ b.example.com.
│  ├─ 0.a.b.example.com.
│  │  ├─ 4.3.2.1.0.a.b.example.com.
│  │  ├─ 4.4.3.2.1.0.a.b.example.com.
│  │  └─ 5.4.3.2.1.0.a.b.example.com.
│  └─ 1.a.b.example.com.
│     ├─ 5.4.3.2.1.a.b.example.com.
│     └─ 6.5.4.3.2.1.a.b.example.com.
└─ c.example.com.
```

And still I think visually showing (grey or whatever) that these are nonexistent RR would be helpful.

Oh, and BTW, is it necessary to show them FQDN each time? Clearer, no?

```
example.com.
├─ 0
├─ b
│  ├─ 0.a.b
│  │  ├─ 4.3.2.1.0.a.b
│  │  ├─ 4.4.3.2.1.0.a.b
│  │  └─ 5.4.3.2.1.0.a.b
│  └─ 1.a.b
│     ├─ 5.4.3.2.1.a.b
│     └─ 6.5.4.3.2.1.a.b
└─ c
```
